### ISSUE TITLE
Can install on Ubuntu 14.10

### ISSUE THREAD

**Issue by changeableface** [olav-st/screencloud#110]
Can't install correctly on 14.10 - worked fine when on 14.04 LTS - updated and this is the only program that broke :(

```
matthew@matthew-ubuntu:~/Downloads$ sudo dpkg -i screencloud_1.1.6-1_amd64.deb 
Selecting previously unselected package screencloud.
(Reading database ... 358267 files and directories currently installed.)
Preparing to unpack screencloud_1.1.6-1_amd64.deb ...
Unpacking screencloud (1.1.6-1) ...
dpkg: dependency problems prevent configuration of screencloud:
 screencloud depends on libqtmultimediakit1; however:
  Package libqtmultimediakit1:amd64 is not installed.
 screencloud depends on libqxt-core0; however:
  Package libqxt-core0:amd64 is not installed.
 screencloud depends on libqxt-gui0; however:
  Package libqxt-gui0:amd64 is not installed.
 screencloud depends on libquazip0; however:
  Package libquazip0:amd64 is not installed.

dpkg: error processing package screencloud (--install):
 dependency problems - leaving unconfigured
Processing triggers for bamfdaemon (0.5.1+14.10.20140925-0ubuntu1) ...
Rebuilding /usr/share/applications/bamf-2.index...
Processing triggers for desktop-file-utils (0.22-1ubuntu2) ...
Processing triggers for mime-support (3.55ubuntu1) ...
Processing triggers for gnome-menus (3.10.1-0ubuntu2) ...
Processing triggers for hicolor-icon-theme (0.13-1) ...
Errors were encountered while processing:
 screencloud
```

Looks like it wants libquazip0 which has been upgrade as part of something else.

**Comment by changeableface** [comment]
Hi, 

Downloaded, still says;

Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies.
 screencloud : Depends: libquazip0 but it is not going to be installed
E: Unable to correct problems, you have held broken packages.

:(

**Comment by boobaa** [comment]
Manually installing libquazip0 removes libquazip1 and marble-plugins (at least for me). After this, I can install screencloud (along with sni-qt which is a dependency of it). However, it would be even better if screencloud had been compiled against the latest stable libraries (libquazip1 this time).

**Comment by deanperry** [comment]
I'm having the same issue with Ubuntu 14.10 :(

**Comment by changeableface** [comment]
FYI - Upgraded to 15.04 - *didn't* expect this to fix anything, and it hasn't, still same issue.

**Comment by changeableface** [comment]
Found a work around for anyone interested;

`$ wget http://download.opensuse.org/repositories/home:/olav-st/xUbuntu_14.04/amd64/screencloud_1.2.0-1_amd64.deb`
`$ sudo ln -s /usr/lib/x86_64-linux-gnu/libquazip.so.1.0.0 /usr/lib/x86_64-linux-gnu/libquazip.so.0`
`$ sudo dpkg -i –force-depends screencloud_1.2.0-1_amd64.deb`
`$ sudo apt-get install -f`

Cobbled together from this ticket; https://github.com/olav-st/screencloud/issues/144

Works fine for me now, but it's obviously not idea.

**Comment by insanesvk** [comment]
Thanks @changeableface :)

**Comment by anjan011** [comment]
It would be nice, if screencloud can update their package with this simple workaround as changeableface  mentioned. I m just glad it works with a simple fix like that.

**Comment by dimkir** [comment]
Tried installing on Ubuntu 14.04.5 latest version of screencloud 1.3 using instructions from the site. 
Installation (via `apt-get install screencloud`) completed successfully but upon running the app  getting this error:

```
user@host:~$ screencloud-1.3.0
screencloud-1.3.0: error while loading shared libraries: libquazip.so.1: cannot open shared object file: No such file or directory
```

The listing of my libquazes is:

```
user@host:/usr/lib/x86_64-linux-gnu$ ls -l libquaz*
lrwxrwxrwx 1 root root     18 Feb 12  2014 libquazip.so.0 -> libquazip.so.0.6.2
lrwxrwxrwx 1 root root     18 Feb 12  2014 libquazip.so.0.6 -> libquazip.so.0.6.2
-rw-r--r-- 1 root root 179560 Feb 12  2014 libquazip.so.0.6.2
```

@changeableface suggests symlinking `libquazip.so.1.0.0` -> `libquazip.so.0` , but in my case:
 - `libquazip.so.0` already exists 
 - `libquazip.so.1.0.0` does not exist

any ideas?

**Comment by samlev** [comment]
I had the (admittedly bad) idea of `ln -s libquazip.so.0.6.2 libquazip.so.1`, which allowed screencloud to boot, but everything was broken (bunch of error pop-ups, none of my connectors work, and the config for them won't open), so... probably not the best idea.

**Comment by dimkir** [comment]
Ok, I managed to find solution for my case of Ubuntu 14.04.05 and believe it should work for other systems with similar symptoms.

TL;DR
---
```
# worked for me for Ubuntu 14.04.05 64bit (amd64)
wget http://launchpadlibrarian.net/233666756/libquazip1_0.7.1-1_amd64.deb
sudo dpkg -i libquazip1_0.7.1-1_amd64.deb
screencloud 
```

How to fix it step by step
---
Seems like this `libquazip` is some kind of Qt wrapper around another zipping library and it should be part of ubuntu and debian. If you look how this package is [listed under ubuntu packages](https://launchpad.net/ubuntu/+source/libquazip), you can see that the latest version of the library (bundled with Xenial Xerus Ubuntu 16.04) is actually not `1.0` but rather `0.7.1` so seems like when symlinking to `libquazip.so.1` - the number `1` is not actually version, but some other ubuntu internal thingy.  Now we know that screencloud wants the latest version  `0.7.1` .

Let's download it and install it on my Ubuntu 14.04.05 64x machine.

1. Go to https://launchpad.net/ubuntu/+source/libquazip and 
choose /Xenial Xerus/
![image](https://cloud.githubusercontent.com/assets/3438798/17855827/96e664b0-6872-11e6-9ccd-1c3b0ff0f5ca.png)


2. On your package page https://launchpad.net/ubuntu/xenial/amd64/libquazip1/0.7.1-1  see //Builds// section and choose amd64 package (my Ubuntu is 64bit if you're on different architecture you have to choose yours)
![image](https://cloud.githubusercontent.com/assets/3438798/17855923/01388726-6873-11e6-8827-f9a50f65af0d.png)


3. On the [Builds] page choose and download appropriate binary. I chose `libquazip1 0.7.1-1`  (and I have no idea what's the difference vs `libquazip5 0.7.1-1`).
![image](https://cloud.githubusercontent.com/assets/3438798/17855948/277c320c-6873-11e6-87c6-a26d5921f8e0.png)


4. On the download page choose to download `libquazip1_0.7.1-1_amd64.deb`
![image](https://cloud.githubusercontent.com/assets/3438798/17856011/67a59576-6873-11e6-9eac-8ef5eeda3ff5.png)

5. As you finished downloading `libquazip1_0.7.1-1_amd64.deb`, install it on your system using
```
sudo dpkg -i libquazip1_0.7.1-1_amd64.deb
```
This will uninstall your older `libquazip.so.0.6.2` (so probably wise to back it up before) and install the new version and create `libquazip.so.1.0.0` required by screencloud.

**Comment by changeableface** [comment]
Hi dimkir, 

Thanks for the in depth fix you've posted. Since I originally posted this message I've swapped OS to OSX so I'm not longer suffering the pain, but no doubt this will help others. 

Matt

**Comment by FatBoyXPC** [comment]
I tried this fix, but I'm unable to get imgur working. I was getting this error:

```
➜  ~ screencloud
"sni-qt/7042" WARN  00:27:56.542 void StatusNotifierItemFactory::connectToSnw() Invalid interface to SNW_SERVICE 
[WARN] "Error while parsing script file /home/james/.local/share/data/screencloud/ScreenCloud/plugins/imgur/main.py" 
[INFO] "Loaded 1 plugin(s)" 
[INFO] "Using bundled icon: :/systray/trayicon-gray.svg" 
[WARN] "Failed to call isConfigured() in ImgurUploader" 
[INFO] "Plugin update check. Has version 1.8 of 'imgur'. Latest is 1.8" 
[INFO] "Plugin update check. Has version 1.8 of 'imgur'. Latest is 1.8" 
Got a buffer underflow! 
[WARN] "Error while playing audio. Code: 3" 
[WARN] "Failed to call showSettingsUI() in ImgurUploader" 
^C[1]    7042 killed     screencloud
➜  ~ 
```
